reference/extensions/hooks.rst

```diff
@@ -1,4 +1,172 @@
 .. _reference_extensions_hooks:
 
 Hooks
------
\ No newline at end of file
+-----
+
+The Conan hooks is a feature intended to extend the Conan functionalities to perform certain orthogonal operations,
+like some quality checks, in different stages of a package creation process, like pre-build and post-build.
+
+Hook structure
+--------------
+
+A hook is a Python function that will be executed at certain points of Conan workflow
+to customize the client behavior without modifying the client sources or the recipe ones.
+
+Hooks can implement any functionality: it could be Conan commands, recipe interactions
+such as exporting or packaging, or interactions with the remotes.
+
+Here is an example of a simple hook:
+
+.. code-block:: python
+   :caption: *hook_example.py*
+
+    import os
+    from conan.tools.files import load
+
+    def pre_export(conanfile):
+        for field in ["url", "license", "description"]:
+            field_value = getattr(conanfile, field, None)
+            if not field_value:
+                conanfile.output.error(f"[REQUIRED ATTRIBUTES] Conanfile doesn't have '{field}'.
+                                          It is recommended to add it as attribute.")
+
+This hook checks the recipe content prior to it being exported and prior to downloading the sources. Basically the
+``pre_export()`` function checks the attributes of the ``conanfile`` object to see if there is an URL, a license and a description and if missing,
+warns the user with a message through the ``conanfile.output``. This is done **before** the recipe is exported to the local cache.
+
+Any kind of Python script can be executed. You can create global functions and call them from different hook functions, import from a
+relative module and warn, error or even raise to abort the Conan client execution.
+
+Other useful task where a hook may come handy are the upload and download actions. There are **pre** and **post** functions for every
+download/upload as a whole and for fine download tasks such as recipe and package downloads/uploads.
```

Comment:
```suggestion
```

Not anymore, the upload/download hooks have been removed, and left only those connected to recipe methods